client/scss/components/_footer.scss

```diff
@@ -103,6 +125,7 @@
 
   .icon {
     width: theme('spacing.4');
+    height: theme('spacing.4');
```

Comment:
Without this, we had this issue (even before this PR) where the svg's height goes beyond the button's, but this wasn't noticeable until we added `height: auto` to the button in the earlier iteration of this PR.

<img width="382" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/fa3db64e-c06f-48b4-9f39-cc330d9aa1e3">